### PR TITLE
duplicate organization and orggroup in core

### PR DESCRIPTION
**Pull request by jansenk**
-Duplicated Organization and OrganizationGroup into core app, created migrations
-Made enrollments app OrganizationGroup parent foreign key explicit due to conflicts with new core model
-Made Program point to new core Organization
-copied tests and factories into core

### CODE REVIEW

File: registrar/apps/core/admin.py [hunk 1]
```diff
@@ -19,4 +24,18 @@ class CustomUserAdmin(UserAdmin):
...
+    date_hierarchy = 'modified'
+
+
+class OrganizationGroupAdmin(admin.ModelAdmin):
+    list_display = ('name', 'organization', 'role')
+    exclude = ('permissions',)
+
+
 admin.site.register(User, CustomUserAdmin)
+admin.site.register(Organization, OrganizationAdmin)
```
Comment (kdmccormick) [inline registrar/apps/core/admin.py:40]: You'll probably want to remove the corresponding `admin.site.register` for the `enrollments` version of the Org models.

File: registrar/apps/core/tests/test_models.py [hunk 2]
```diff
@@ -43,3 +49,57 @@ def test_string(self):
         full_name = 'Bob'
         user = G(User, full_name=full_name)
         self.assertEqual(str(user), full_name)
+
+
+@ddt.ddt
+class OrganizationGroupTests(TestCase):
```
Comment (kdmccormick) [inline registrar/apps/core/tests/test_models.py:55]: Please remove the `enrollments` version of these tests.

File: registrar/apps/core/tests/factories.py [hunk 3]
```diff
@@ -32,3 +35,32 @@ class Meta:
 
 class StaffUserFactory(UserFactory):
     is_staff = True
+
+
+def name_to_key(name):
```
Comment (kdmccormick) [inline registrar/apps/core/tests/factories.py:40]: Please remove the stuff you added here from the `enrollments` factory module.

**Review by kdmccormick — CHANGES_REQUESTED** [review]
There are a few more references to the old Org models. Let me know when you remove those and I'll give a re-review.

File: registrar/apps/enrollments/models.py [hunk 4]
```diff
@@ -71,6 +72,7 @@ class Meta(object):
         (perms.OrganizationReadWriteEnrollmentsRole.name, 'Read and Write Enrollments Data'),
     )
 
+    group_ptr = models.OneToOneField(Group, parent_link=True, related_name="organization_old")
```
Comment (schenedx) [inline registrar/apps/enrollments/models.py:75]: Why do we need this new field?
I thought we just need to remove this model right? Keep in mind, we remove this model, but not create the migration file will help us avoid the deployment error with this change.

File: registrar/apps/enrollments/migrations/0004_prepare_remove_organization.py [hunk 5]
```diff
@@ -0,0 +1,26 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.18 on 2019-03-13 20:49
+from __future__ import unicode_literals
```
Comment (schenedx) [inline registrar/apps/enrollments/migrations/0004_prepare_remove_organization.py:3]: I really don't think this migration file is needed.

**Review by schenedx — CHANGES_REQUESTED** [review]
I think you can go a lot further in removing code.

**Review by kdmccormick — APPROVED** [review]
LGTM :+1:

**Review by schenedx — APPROVED** [review]
👍

**Pull request merged**